libslide/src/emit.rs

```diff
@@ -148,6 +148,25 @@ macro_rules! latex_wrap {
     };
 }
 
+fn join_emits<E: Emit>(list: &[E], emit: impl FnMut(&E) -> String) -> String {
+    list.iter().map(emit).collect::<Vec<_>>().join("\n")
+}
+
+fmt_emit_impl!(StmtList);
+impl Emit for StmtList {
+    fn emit_pretty(&self, config: EmitConfig) -> String {
+        join_emits(&self.list, |s| s.emit_pretty(config))
+    }
```

Comment:
can you explain the closure here. and why your passing it to a map? I just understand it

Reply:
The StmtList is a list of statements. We are passing a closure to describe how each statement should be emitted, then mapping each statement with that emit fn and joining into a string..